channels/offline-channel-js/src/OfflineChannel.ts

```diff
@@ -0,0 +1,604 @@
+            }
+
+
+            function _getPayload(evt: IPostTransmissionTelemetryItem | ITelemetryItem) {
+                try {
+
+                    if (evt) {
+                        evt.iKey = evt.iKey || _urlCfg.iKey;
+                        return _offineSupport.serialize(evt, _convertUndefined);
+                    }
+                    
+                } catch (e) {
+                    // eslint-disable-next-line no-empty
+                }
+                return null;
+                
+            }
+
+            function _constructPayloadData(payloadArr: string[], criticalCnt?: number) {
+
+                try {
+                    if (!_offineSupport) {
+                        return null;
+                    }
+                    let cnt = criticalCnt || 0;
+                    let payload = _offineSupport.batch(payloadArr);
+                    let payloadData: IPayloadData = {
+                        urlString: _urlCfg.url,
+                        data: payload,
+                        headers: {},
+                        criticalCnt: cnt
+                    } as IStorageTelemetryItem;
+                    let details = _offineSupport.getOfflineRequestDetails && _offineSupport.getOfflineRequestDetails();
+                    if (details) {
+                        let url = details.url;
+                        if (url) {
+                            payloadData.urlString = url;
+                        }
+                        let hdrs = details.hdrs;
+                        if (hdrs) {
+                            payloadData.headers = hdrs;
+                        }
+                    }
+                    return payloadData;
+
+                } catch(e) {
+                    // eslint-disable-next-line no-empty
+                }
+                return null;
+            }
+
+            function _createUrlConfig(coreConfig: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?: ITelemetryPluginChain) {
+
+                _self._addHook(onConfigChange(coreConfig, (details) => {
+                    let storageConfig: ILocalStorageConfiguration = null;
+                    let theConfig = details.cfg;
+
+                    let ctx = createProcessTelemetryContext(null, theConfig, core);
+                    storageConfig = ctx.getExtCfg<ILocalStorageConfiguration>(_self.identifier, defaultLocalStorageConfig);
+                    let channelIds = storageConfig.primaryOnlineChannelId;
+                    if (channelIds && channelIds.length) {
+                        arrForEach(channelIds, (id) => {
+                            let plugin = _self.core.getPlugin<IChannelControls>(id);
+                            let channel = plugin && plugin.plugin;
+                            if (channel) {
+                                _primaryChannelId = id;
+                                _offineSupport = channel.getOfflineSupport();
```

Comment:
Someone (might) try and use this with an older version of Application Insights at which point the `getOfflineSupport` function won't exist and will cause this to throw an exception, so we need to check if the function is there before calling.